src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -194,6 +197,14 @@ protected ExtensionsRunner(Extension extension) throws IOException {
         logger.info("SSL is " + sslText + " for transport");
         this.settings = settingsBuilder.build();
 
+        if (extensionSettings.getShortExtensionName() != null) {
+            // initialize ExtensionRouteHandlerFactory
+            ExtensionRouteHandlerFactory factory = ExtensionRouteHandlerFactory.getInstance();
+            if (!factory.isInitialized()) {
```

Comment:
I'm trying to think of a situation where this would ever be initialized before the ExtensionsRunner constructor was called.  Possible but someone would have to go out of their way to do it.   

Not sure we need a factory at all, and this is just one more reason this solution is overly complex.